packages/blocks/src/store/reducer.js

```diff
@@ -389,6 +392,11 @@ export function blockBindingsSources( state = {}, action ) {
 			return {
 				...state,
 				[ action.name ]: {
+					/*
+					 * Keep the exisitng properties in case the source has been registered
+					 * in the client before bootstrapping.
+					 */
+					...state[ action.name ],
 					label: action.label,
 					usesContext: action.usesContext,
```

Comment:
We'll need to reuse the logic from the [context PR](https://github.com/WordPress/gutenberg/pull/63513/files#diff-d5aba57655b6c1abcbd8f937a61ad98110057a6ae09f2db596811dca31c54030R1544-R1549) here to make sure the `usesContext` values get merged properly and the server doesn't overwrite the client values.

Reply:
You're right there will be "conclict changes" once any of the mentioned pull requests gets merged. I don't have a strong opinion of which one should go first, I believe it is irrelevant. I will rebase and resolve the issue once it gets merged.